### PR TITLE
Fix rose tests for GMT 6.2.0

### LINKED ISSUE #1315: Some rose tests are wrong
**Description of the problem**

_The issue was first reported in https://github.com/GenericMappingTools/gmt/pull/5289#issuecomment-852667972._

The GMT `rose` module expects *length* and *azimuth*. 

The sample data `@fractures_06.txt` contains *azimuth*,*length*. When writing the `load_fractures_compilation` function, we decided to return *length*,*azimuth*, rather than *azimuth*,*length*, so that the rose tests don't need to swap the columns. However, when writing the tests, we incorrectly used `incols=[1, 0]`, which swaps the columns to *azimuth*,*length* again. 

The baseline images are correct because the **-i** option doesn't work as expected due to an upstream bug (fixed in https://github.com/GenericMappingTools/gmt/pull/5289). Thus, after the fix, the `incols=[1, 0]` parameter is no longer needed.

We need to fix these rose tests after bumping to GMT 6.2.0.

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Fixes #1315 

Address #1320.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_rose.py [hunk 1]
```diff
@@ -198,10 +193,16 @@ def test_rose_deprecate_columns_to_incols(data_fractures_compilation):
 
     Modified from the test_rose_bools() test.
     """
+
+    data = data_fractures_compilation
+    # swap data columns, below the use of the columns parameter
+    # reverses this action
+    data = data[["azimuth", "length"]]
```
Comment (seisman) [inline pygmt/tests/test_rose.py:200]: Does this work?
```suggestion
    # swap data columns, below the use of the columns parameter
    # reverses this action
    data = data_fractures_compilation[["azimuth", "length"]]
```

Comment (michaelgrund) [reply pygmt/tests/test_rose.py:200]: Should work, yes!

**Review by seisman — APPROVED** [review]

File: pygmt/tests/test_rose.py [hunk 2]
```diff
@@ -141,17 +141,15 @@ def test_rose_plot_with_transparency(data_fractures_compilation):
 @pytest.mark.mpl_image_compare
 def test_rose_no_sectors(data_fractures_compilation):
     """
-    Test supplying a data file containing a list of fracture lengths and
-    azimuth as digitized from geological maps to the data argument (lengths are
-    stored in the second column, azimuths in the first, specify via columns).
+    Test supplying the sample fractures compilation dataset to the data
+    argument.
```
Comment (weiji14) [inline pygmt/tests/test_rose.py:145]:
```suggestion
    parameter.
```

File: pygmt/tests/test_rose.py [hunk 3]
```diff
@@ -165,9 +163,8 @@ def test_rose_no_sectors(data_fractures_compilation):
 @pytest.mark.mpl_image_compare
 def test_rose_bools(data_fractures_compilation):
     """
-    Test supplying a data file containing a list of fracture lengths and
-    azimuth as digitized from geological maps to the data argument (lengths are
-    stored in the second column, azimuths in the first, specify via columns).
+    Test supplying the sample fractures compilation dataset to the data
+    argument.
```
Comment (weiji14) [inline pygmt/tests/test_rose.py:167]:
```suggestion
    parameter.
```

File: pygmt/tests/test_rose.py [hunk 4]
```diff
@@ -115,9 +117,8 @@ def test_rose_plot_data_using_cpt(data):
 @pytest.mark.mpl_image_compare
 def test_rose_plot_with_transparency(data_fractures_compilation):
     """
-    Test supplying a data file containing a list of fracture lengths and
-    azimuth as digitized from geological maps to the data argument (lengths are
-    stored in the second column, azimuths in the first, specify via columns).
+    Test supplying the sample fractures compilation dataset to the data
+    argument.
```
Comment (weiji14) [inline pygmt/tests/test_rose.py:121]:
```suggestion
    parameter.
```

File: pygmt/tests/test_rose.py [hunk 5]
```diff
@@ -198,10 +194,15 @@ def test_rose_deprecate_columns_to_incols(data_fractures_compilation):
 
     Modified from the test_rose_bools() test.
     """
+
+    # swap data columns of the sample fractures compilation dataset, below the
+    # use of the columns parameter reverses this action
```
Comment (weiji14) [inline pygmt/tests/test_rose.py:199]:
```suggestion
    # swap data column order of the sample fractures compilation dataset,
    # as the use of the 'columns' parameter will reverse this action
```

**Review by weiji14 — APPROVED** [review]
Just a few nitpicks, see #886 and https://docs.python.org/3.9/faq/programming.html#faq-argument-vs-parameter re: arguments vs parameters.

**Comment by weiji14** [comment]
Feel free to merge @michaelgrund :rocket:

**Pull request merged**